lib/shopify_app/session_storage.rb

```diff
@@ -0,0 +1,23 @@
+module ShopifyApp
+  module SessionStorage
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def store(session)
+        shop = self.find_or_initialize_by(shopify_domain: session.url)
+        shop.shopify_token = session.token
+        shop.save!
+        shop.id
+      end
+
+      def retrieve(id)
+        return unless id
+        shop = self.find(id)
+        ShopifyAPI::Session.new(shop.shopify_domain, shop.shopify_token)
+      rescue ActiveRecord::RecordNotFound
```

Comment:
I might prefer something like:

```ruby
if shop = self.find_by(id: id) # won't raise if not found
  ShopifyAPI::Session.new(shop.shopify_domain, shop.shopify_token)
end
```


… just so we don't need to `rescue` here. But up to you.